samples/mpu_stack_guard_test/testcase.ini

```diff
@@ -0,0 +1,10 @@
+[test]
+build_only = true
+tags = apps
+
+[test_stack_guard]
+build_only = true
```

Comment:
why is this a build_only test?
There are ways of testing whether the system correctly caught an exception.
See tests/kernel/fatal for an example.
I'd like to see this moved to tests/ and verify correctness at runtime.